lib/tests/test_revset.rs

```diff
@@ -1178,6 +1178,33 @@ fn test_evaluate_expression_ancestors(use_git: bool) {
             root_commit.id().clone(),
         ]
     );
+
+    // Can find last n ancestors of a commit
+    assert_eq!(
+        resolve_commit_ids(
+            mut_repo,
+            &format!("ancestors({}, 0)", commit2.id().hex())
+        ),
+        vec![]
+    );
```

Comment:
I thinks there are pros and cons. In Mercurial, `depth` parameter is defined as the maximum generation. I thought it would be intuitive, but I was struggling with adding the min parameter. `ancestors(x, max_depth, min_depth)` doesn't make sense, but we can't reorder these depth parameters. OTOH, if we defined the `depth` as a length parameter, the function could be extended to `ancestors(x, depth, offset=0)`, meaning ancestors within the range `offset..(offset + depth)`.

FWIW, I'm not sure whether an integer range `m..n` should be inclusive. The set `x..y` is exclusive/inclusive, but such definition is useless in general.